### PR TITLE
Bug 1797573: Add pod count on dev-console monitoring dashboard page

### PR DESCRIPTION
**Pull request by vikram-raj**
Bug fix - https://issues.redhat.com/browse/ODC-2835

Screenshot from 2020-01-27 15-17-42

### CODE REVIEW

File: frontend/packages/dev-console/src/components/monitoring/dashboard/MonitoringDashboardPodCount.tsx [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+
+const PodCount: React.FC<PodCountProps> = ({ pods }) => {
+  return (
+    <div className="odc-monitoring-dashboard-pod-count">
+      <h5>Pod Count</h5>
+      <Bullseye>
+        {pods.loaded ? (
+          <h1 className="odc-monitoring-dashboard-pod-count__count">{pods.data.length}</h1>
+        ) : (
+          <LoadingInline />
```
Comment (invincibleJai) [inline frontend/packages/dev-console/src/components/monitoring/dashboard/MonitoringDashboardPodCount.tsx]: @vikram-raj can we have specs for conditional branch here?

Comment (vikram-raj) [reply frontend/packages/dev-console/src/components/monitoring/dashboard/MonitoringDashboardPodCount.tsx]: Adding test for this.

Comment (vikram-raj) [reply frontend/packages/dev-console/src/components/monitoring/dashboard/MonitoringDashboardPodCount.tsx]: @invincibleJai Added tests for this. PTAL.

File: frontend/packages/dev-console/src/components/monitoring/dashboard/MonitoringDashboardPodCount.tsx [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+
+interface MonitoringDasboardPodCountProps {
+  namespace: string;
+}
+
+interface PodCountProps {
+  pods?: FirehoseResult<PodKind[]>;
+}
+
+export const PodCount: React.FC<PodCountProps> = ({ pods }) => {
```
Comment (invincibleJai) [inline frontend/packages/dev-console/src/components/monitoring/dashboard/MonitoringDashboardPodCount.tsx]: We usually have one component per file, will it make sense here as well?  I know here `MonitoringDasboardPodCount`  we just have `Firehose`. WDYT?

File: frontend/packages/dev-console/src/components/monitoring/dashboard/MonitoringDashboard.tsx [hunk 3]
```diff
@@ -22,11 +22,11 @@ const MonitoringDashboard: React.FC<MonitoringDashboardProps> = ({ match }) => {
...
       <Grid className="co-m-pane__body" gutter="md">
-        <GridItem span={4} rowSpan={1}>
-          <MonitoringDasboardCountBlock />
+        <GridItem span={3} rowSpan={1}>
+          <MonitoringDasboardPodCount namespace={namespace} />
         </GridItem>
-        {_.map(queries, (q) => (
-          <GridItem span={4} key={q.title}>
+        {_.map(queries, (q, i) => (
+          <GridItem span={i === 0 ? 5 : 4} key={q.title}>
```
Comment (invincibleJai) [inline frontend/packages/dev-console/src/components/monitoring/dashboard/MonitoringDashboard.tsx]: any reason for this condition here?

File: frontend/packages/dev-console/src/components/monitoring/dashboard/MonitoringDashboardPodCount.tsx (hunk 2 above)
Comment (vikram-raj) [reply frontend/packages/dev-console/src/components/monitoring/dashboard/MonitoringDashboardPodCount.tsx]: In ```MonitoringDasboardPodCount``` just have ```Firehose``` that is why put it in one file.

File: frontend/packages/dev-console/src/components/monitoring/dashboard/MonitoringDashboard.tsx (hunk 3 above)
Comment (vikram-raj) [reply frontend/packages/dev-console/src/components/monitoring/dashboard/MonitoringDashboard.tsx]: Yes, as per UX podCount width is smaller than other block and the first graph block is bigger than others.

File: frontend/packages/dev-console/src/components/monitoring/dashboard/MonitoringDashboardGraph.tsx [hunk 4]
```diff
@@ -45,13 +42,18 @@ const MonitoringDashboardGraph: React.FC<MonitoringDashboardGraphProps> = ({
...
+          <PrometheusGraphLink query={query}>
+            <QueryBrowser
+              hideControls
+              defaultTimespan={defaultTimespan}
+              namespace={namespace}
+              queries={[query]}
+              isStack
+            />
+          </PrometheusGraphLink>
+        </>
```
Comment (invincibleJai) [inline frontend/packages/dev-console/src/components/monitoring/dashboard/MonitoringDashboardGraph.tsx]: @vikram-raj   can we avoid repetition of `PrometheusGraphLink` and `QueryBrowser` and just change `isStack` to true or false

Comment (vikram-raj) [reply frontend/packages/dev-console/src/components/monitoring/dashboard/MonitoringDashboardGraph.tsx]: Done.

File: frontend/packages/dev-console/src/components/monitoring/dashboard/__tests__/MonitoringDashboardGraph.spec.tsx [hunk 5]
```diff
@@ -23,7 +22,7 @@ describe('Monitoring Dashboard graph', () => {
 
   it('should render area graph', () => {
     const wrapper = shallow(<MonitoringDashboardGraph {...monitoringDashboardGraphProps} />);
-    expect(wrapper.find(Area).exists()).toBe(true);
+    expect(wrapper.find(QueryBrowser).props().isStack).toBe(true);
```
Comment (invincibleJai) [inline frontend/packages/dev-console/src/components/monitoring/dashboard/__tests__/MonitoringDashboardGraph.spec.tsx]: @vikram-raj  can we extend the test to check `isStack` as well in props for normal series and stacked chart?

Comment (vikram-raj) [reply frontend/packages/dev-console/src/components/monitoring/dashboard/__tests__/MonitoringDashboardGraph.spec.tsx]: Here is a test case for render the line graph https://github.com/openshift/console/pull/4072/files#diff-a3f715c19539634ce725a50299562c84R31

Comment (invincibleJai) [reply frontend/packages/dev-console/src/components/monitoring/dashboard/__tests__/MonitoringDashboardGraph.spec.tsx]: Thanks @vikram-raj

File: frontend/packages/dev-console/src/components/monitoring/dashboard/MonitoringDashboardGraph.tsx [hunk 6]
```diff
@@ -27,32 +26,21 @@ const MonitoringDashboardGraph: React.FC<MonitoringDashboardGraphProps> = ({
...
-      ) : (
-        <Area
-          title={title}
-          humanize={humanize}
-          byteDataType={byteDataType}
-          namespace={namespace}
-          query={query}
-        />
-      )}
+      <>
```
Comment (christianvogt) [inline frontend/packages/dev-console/src/components/monitoring/dashboard/MonitoringDashboardGraph.tsx]: unnecessary fragment

File: frontend/packages/dev-console/src/components/monitoring/dashboard/MonitoringDashboardPodCount.tsx (hunk 2 above)
Comment (christianvogt) [reply frontend/packages/dev-console/src/components/monitoring/dashboard/MonitoringDashboardPodCount.tsx]: I'm ok with multiple components per file if it's used internally. The idea being that only the default export should ever be consumed elsewhere.

File: frontend/packages/dev-console/src/components/monitoring/dashboard/MonitoringDashboardGraph.tsx (hunk 6 above)
Comment (vikram-raj) [reply frontend/packages/dev-console/src/components/monitoring/dashboard/MonitoringDashboardGraph.tsx]: removed it.

**Comment by vikram-raj** [comment]
/retitle Bug 1797573: Add pod count on dev-console monitoring dashboard page

**Pull request merged**